internal/domain/product.go

```diff
@@ -0,0 +1,60 @@
+package domain
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+var ProductCategory = []string{"Clothing", "Accescories", "Footware", "Beverages"}
+
+type Product struct {
+	ID          string    `db:"id"`
+	CreatedAt   time.Time `db:"created_at"`
+	Name        string    `db:"name"`
+	Sku         string    `db:"sku"`
+	Category    string    `db:"category"`
+	ImageUrls   []string  `db:"image_urls"`
+	Notes       string    `db:"notes"`
+	Price       int       `db:"price"`
+	Stock       int       `db:"stock"`
+	Location    string    `db:"stock"`
+	IsAvailable bool      `db:"is_available"`
+}
+
+type CreateProductRequest struct {
+	Name        string   `json:"name"`
+	Sku         string   `json:"sku"`
+	Category    string   `json:"category"`
+	ImageUrls   []string `json:"imageUrls"`
+	Notes       string   `json:"notes"`
+	Price       int      `json:"price"`
+	Stock       int      `json:"stock"`
+	Location    string   `json:"location"`
+	IsAvailable bool     `json:"isAvailable"`
+}
+
+type CreateProductResponse struct {
+	ID        string    `json:"id"`
+	CreatedAt time.Time `json:"createdAt"`
+}
+
+func NewProductFromDTO(body CreateProductRequest) Product {
```

Comment:
oh iya lupa. dijadiin struct CreateProductRequest aja mar. jadi bukan function biasa

```suggestion
func (c CreateProductRequest) NewProductFromDTORequest() Product {
```

Reply:
no, passing data aja, lbh sesuai krn ini golang

Reply:
wkwk. gak ada hubungannya sama golang

Reply:
klo gitu buat helper mar. jgn disimpan di domain mar